api/providers/views.py

```diff
@@ -792,3 +803,79 @@ def get_default_queryset(self):
 
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+
+class RegistrationBulkCreate(APIView):
+    parser_classes = [FileUploadParser]
+
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        CanAddModerator,
+    )
+
+    def get_hash(self, file_obj):
+        BLOCK_SIZE = 65536
```

Comment:
I can do 2**16. There is no reason for the value. I chose it because it seems to be a good chunk size for a file under 1mb. I could change the chunk size to to other values. We could do 2**20, which will take the entire file in exactly one chunk, if memory is not an issue.